### ISSUE TITLE
Generating pdf with LuaLateX

### ISSUE THREAD

**Issue by flagarde** [doxygen/doxygen#10147]
**Describe the bug**
Generating LaTeX refman is not working anymore

**To Reproduce**
TexLive 2021 on Linux, Doxygen 1.9.7. Just running basic refman generate this error with LuaLaTeX
This is LuaHBTeX, Version 1.13.2 (TeX Live 2021) 
 system commands enabled.

(/home/runner/work/cpp-terminal/cpp-terminal/build/docs/doxygen/latex/refman.te
x
LaTeX2e <2021-11-15> patch level 1
tput: No value for $TERM and no -T specified
 L3 programming layer <2022-02-24>
! Undefined control sequence.
l.4   \pdfminorversion
                    =7
? 
! Emergency stop.
l.4   \pdfminorversion
                    =7

**Comment by albert-github** [comment]
The problem will probably be a general problem, but especially the settings in the `Doxyfile`are essential (probably a project with your `Doxyfile`and an empty àa.h`file is sufficient), so
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by flagarde** [comment]
Sure, I will try to provide this.

**Comment by albert-github** [comment]
I just did a small test with:
- empty file `aa.h`
- `Doxyfile` with just: `LATEX_CMD_NAME         = lualatex`
- run `doxygen`
- run in the latex directory : `make`


This results in the same warning. I also get the warning with xelatex.
(no test case further necessary)

**Comment by albert-github** [comment]
I already found some starting points for a fix in https://tex.stackexchange.com/a/374993/44119:

> As of LuaTeX v0.87, the pdfTeX-derived extension primitives are renamed. Here, you want
> 
> `\pdfvariable minorversion`
> 
> The details of emulation are given in the LuaTeX manual, or you can load the luatex85 package which wraps up this emulation for the transition.


For xelatex I found https://tex.stackexchange.com/a/536258/44119:

> You can use a \special to set the version:
> `
> \documentclass{article}
> %\special{dvipdfmx:config z 0} %to uncompress and check
> \special{pdf:minorversion 7} %set minorversion
> 
> 
> \begin{document}
> abc
> \end{document}
>`

**Comment by flagarde** [comment]
I modified a bit the header.sty but then I have some problems with commands generated by doxygen itselft https://github.com/doxygen/doxygen/blob/396688347198e076597050259f4835853bdc3b44/src/latexgen.cpp#L2217-L2240

```bash
Writing index file refman.idx

(/home/travel/.miktex/texmfs/install/tex/latex/newunicodechar/newunicodechar.st
y)
! Undefined control sequence.
\doxynewunicodechar ...uc@tempa \@nil }\nuc@check 
                                                  \if@tempswa \@namedef {u8:...

l.154   \doxynewunicodechar{⁻}{${}^{-}$}
                                        % Superscript minus
? 
! Undefined control sequence.
\doxynewunicodechar ...uc@tempa \@nil }\nuc@check 
                                                  \if@tempswa \@namedef {u8:...

l.155   \doxynewunicodechar{²}{${}^{2}$}
                                       % Superscript two
? 
! Undefined control sequence.
\doxynewunicodechar ...uc@tempa \@nil }\nuc@check 
                                                  \if@tempswa \@namedef {u8:...

l.156   \doxynewunicodechar{³}{${}^{3}$}
                                       % Superscript three

```

Not sure it is necessary with luatex or xetex

**Comment by albert-github** [comment]
When fixing the original issue I will have a  bit deeper look at this as well, might be that it is not necessary or problematic when not using pdflatex

**Comment by flagarde** [comment]
I can provide a PR for the issue if you want

**Comment by albert-github** [comment]
Thanks, i think it might also be related to the used engine, so should both be fixed in one go.

**Comment by flagarde** [comment]
I'm not sure it's necessary for lualatex and xetex maybe the commands should be skipped for now and added some more if needed in the future ? I can put it on the PR

**Comment by albert-github** [comment]
These are indeed some special characters and we have to see, by means of a small test (also in formulas), what the effect of these characters is with the different engines.

(note with PR I assume you mean problem report? on GitHub it is quite common to refer to a pull request as PR though).

**Comment by flagarde** [comment]
I mean pull request

**Comment by acodecow** [comment]
> I modified a bit the header.sty but then I have some problems with commands generated by doxygen itselft
> 
> https://github.com/doxygen/doxygen/blob/396688347198e076597050259f4835853bdc3b44/src/latexgen.cpp#L2217-L2240
> 
> ```shell
> Writing index file refman.idx
> 
> (/home/travel/.miktex/texmfs/install/tex/latex/newunicodechar/newunicodechar.st
> y)
> ! Undefined control sequence.
> \doxynewunicodechar ...uc@tempa \@nil }\nuc@check 
>                                                   \if@tempswa \@namedef {u8:...
> 
> l.154   \doxynewunicodechar{⁻}{${}^{-}$}
>                                         % Superscript minus
> ? 
> ! Undefined control sequence.
> \doxynewunicodechar ...uc@tempa \@nil }\nuc@check 
>                                                   \if@tempswa \@namedef {u8:...
> 
> l.155   \doxynewunicodechar{²}{${}^{2}$}
>                                        % Superscript two
> ? 
> ! Undefined control sequence.
> \doxynewunicodechar ...uc@tempa \@nil }\nuc@check 
>                                                   \if@tempswa \@namedef {u8:...
> 
> l.156   \doxynewunicodechar{³}{${}^{3}$}
>                                        % Superscript three
> ```
> 
> Not sure it is necessary with luatex or xetex

revert a PR change solve it for me:

https://github.com/doxygen/doxygen/pull/9981/files?diff=unified&w=0#diff-9ed350162863756e0fd534f724ae6a06bac9120df53286b9472486d935a51cb2L2175-L2178

**Comment by albert-github** [comment]
@acodecow 
- Did you revert the entire PR or just part of it? (just the part in latexgen.cpp?)

**Comment by albert-github** [comment]
@flagarde In the comments we were talking about providing a Pull Request, I can see that you created a fix in your forked repository, do you intend to make an official pull request out of it?

**Comment by flagarde** [comment]
@albert-github I can try to make one but I will have to dig into the code again as is quite long time since I did the changes.

**Comment by albert-github** [comment]
@flagarde 
Thanks for your answer. Alternative is that I take over by using your code https://github.com/flagarde/doxygen/commit/8d9dba4fffb07b792797a748811c1956c95b0f98 (the etoc_doxygen.sty should not be updated as the differences are just spaces that should have no effect and the `[0` has already been removed.)

What is your preference?

**Comment by flagarde** [comment]
@albert-github feel free to use the code provided if you think it could be useful

**Comment by albert-github** [comment]
@flagarde OK I will have a look at it, I think the code will be useful (and at least give a direction).

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10788

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by flagarde** [comment]
@albert-github Nice ! Thanks a lot !

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
